File: app/src/main/java/com/sortedqueue/learnandroid/dashboard/DashboardNavigationListener.java
package com.sortedqueue.learnandroid.dashboard;

/**
 * Created by Alok on 05/06/17.
 */

public interface DashboardNavigationListener {
    void loadDashboardFragment();
    void loadTopicFragment();
    void loadPresentationFragment(String mainTitle, String topic, int topicIndex, String[] topicArray);
    String getCurrentTopic();
    String getCurrentMainTitle();
    void onNavigateBack();
    void showNavigateToNextTopic();
}

File: app/src/main/java/com/sortedqueue/learnandroid/dashboard/DashboardNavigationListener.kt
package com.sortedqueue.learnandroid.dashboard

/**
 * Created by Alok on 05/06/17.
 */

interface DashboardNavigationListener {
    fun loadDashboardFragment()
    fun loadTopicFragment()
    fun loadPresentationFragment(mainTitle: String, topic: String, topicIndex: Int, topicArray: Array<String>)
    fun onNavigateBack()
    fun showNavigateToNextTopic()
    fun hideNavigateToLayout()
}
